src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -242,6 +252,19 @@ private FormData prepareFormDataSet(FormFields form) {
 				currentList = submitFields;
 			}
 			for (String value : getValues(field)) {
```

Comment:
Disregard last comment. All fields are now being passed to the generator after moving logic from getValues method, and are returning default values as they should be. Please review our movement of the logic and let us know if it needs revision.